src/metalperformanceshaders.cs

```diff
@@ -1135,5 +1541,1851 @@ interface MPSMatrixMultiplication {
+
+		[Export ("initWithDevice:info:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device, MPSImageKeypointRangeInfo info);
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+
+		[Export ("encodeToCommandBuffer:sourceTexture:regions:numberOfRegions:keypointCountBuffer:keypointCountBufferOffset:keypointDataBuffer:keypointDataBufferOffset:")]
+		void Encode (IMTLCommandBuffer commandBuffer, IMTLTexture source, MTLRegion regions, nuint numberOfRegions, IMTLBuffer keypointCountBuffer, nuint keypointCountBufferOffset, IMTLBuffer keypointDataBuffer, nuint keypointDataBufferOffset);
+	}
+	
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSBinaryImageKernel))]
+	interface MPSImageArithmetic {
+		[Export ("primaryScale")]
+		float PrimaryScale { get; set; }
+
+		[Export ("secondaryScale")]
+		float SecondaryScale { get; set; }
+
+		[Export ("bias")]
+		float Bias { get; set; }
+
+		[Export ("primaryStrideInPixels", ArgumentSemantic.Assign)]
+		MTLSize PrimaryStrideInPixels { get; set; }
+
+		[Export ("secondaryStrideInPixels", ArgumentSemantic.Assign)]
+		MTLSize SecondaryStrideInPixels { get; set; }
+
+		//inlining ctor from base class
+		[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageArithmetic))]
+	interface MPSImageAdd {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageArithmetic))]
+	interface MPSImageSubtract {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageArithmetic))]
+	interface MPSImageMultiply {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageArithmetic))]
+	interface MPSImageDivide {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSUnaryImageKernel))]
+	interface MPSImageScale {
+		[Export ("initWithDevice:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (IMTLDevice device);
+
+		[NullAllowed, Export ("scaleTransform", ArgumentSemantic.Assign)]
+		MPSScaleTransform ScaleTransform { get; set; }
+
+		[Export ("initWithCoder:device:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSCoder aDecoder, IMTLDevice device);
+	}
+
+	[TV (11,0), Mac (10, 13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof(MPSImageScale))]
```

Comment:
DisableDefaultCtor

Reply:
I'm also able to create the MPSImageScale subclasses in Xcode with alloc/init and build/run with no errors.